File: source/libocto/content.ts
import type { Content } from "./llm-ir.ts";

export function textContent(content: string): Content["content"] {
  return [{ type: "text", content }];
}

export function contentToText(content: Content["content"]): string {
  return content
    .map(part => {
      if (part.type === "text") return part.content;
      return `Image file: ${part.image.filePath}`;
    })
    .join("\n");
}

File: crates/octofwen-llm/src/ir/content.rs
#[derive(Clone, Debug, PartialEq)]
pub enum ContentPart {
    Text { content: String },
    Image { image: ImageInfo },
}

#[derive(Clone, Debug, Eq, PartialEq)]
pub struct ImageInfo {
    pub file_path: String,
    pub mime_type: String,
    pub base64_data: String,
    pub data_url: String,
    pub size_bytes: Option<u64>,
}
